### ISSUE TITLE
template_preprocess_username trims user's names very strictly (20 chars)

### ISSUE THREAD

**Issue by indigoxela** [backdrop/backdrop-issues#4578]
**Description of the bug**

When a user (node author) has a name longer than 20 characters, its display gets trimmed to 16 characters + three dots.

20 chars is really restrictive (why so short?) and trimming people's names is a bit impolite anyway.

And here's [the code](https://github.com/backdrop/backdrop/blob/8b17a75562d6d18d23b814bf046d8d8eeef2e37f/core/modules/user/user.theme.inc#L277).

**Steps To Reproduce**

1. Enable author name display in a content type
2. Add a user with a name longer than 20 characters
3. View the node

**Actual behavior**

User "Firstname Lastname-Andmore" gets truncated to "Firstname Lastna..."

**Expected behavior**

At least not such a _radical_ truncation. What's the reason for this? Why 20?

FTR: Drupal 7 and 9 do the same - not that this explained, why...

**Comment by klonos** [comment]
@indigoxela doesn't the inline comment above the line of code explain things sufficiently?:

```php
  // Set the name to a formatted name that is safe for printing and
  // that won't break tables by being too long. Keep an unshortened,
  // unsanitized version, in case other preprocess functions want to implement
  // their own shortening logic or add markup. If they do so, they must ensure
  // that $variables['name'] is safe for printing.
```

...is the workaround provided not enough? (using `$variables['name_raw']`)

**Comment by klonos** [comment]
https://stackoverflow.com/questions/30485/what-is-a-reasonable-length-limit-on-person-name-fields

Assuming that people have thought of international names etc.:

> UK Government Data Standards Catalogue suggests 35 characters for each of Given Name and Family Name, or 70 characters for a single field to hold the Full Name.

Having said that, how the name is stored in the db and how it's being rendered on display is a different story. ...and if you think the number you chose is "safe", then...

https://en.wikipedia.org/wiki/Hubert_Blaine_Wolfeschlegelsteinhausenbergerdorff_Sr 🙄 

> is the abbreviated name of a German-born American typesetter who has held the record for the longest personal name ever used. Hubert's name is made up from 27 names. Each of his 26 given names starts with a different letter of the English alphabet in alphabetical order; these are followed by an enormously long single-word surname.

https://www.guinnessworldrecords.com/world-records/67285-longest-personal-name

>The longest name to appear on a birth certificate is for Rhoshandiatellyneshiaunneveshenk Koyaanisquatsiuth Williams, born to Mr and Mrs James Williams in Beaumont, Texas, USA on 12 Sep 1984.

**Comment by indigoxela** [comment]
`$variables['name_raw']` is not available in node templates. I'm using `check_plain($node->name);` in the node template instead.

I'm still not convinced that `20` really makes sense. We don't truncate node titles or terms or menu items that way and these also fit into tables.

Anyway, if you think that this radical truncation is acceptable, feel free to close this issue.

**Comment by indigoxela** [comment]
FTR: the database column "name" in the user table is varchar(60), the "Username" form item has a maxlength of 60.

In contrast to that 20 for display is sparse.

Don't get me wrong, I don't suggest to allow "unlimited", I just feel like `20` isn't plausible.

**Comment by klonos** [comment]
>...if you think that this radical truncation is acceptable, feel free to close this issue.

Nope, I did not say that; I also find it odd. People can trim that on the theme level or with CSS `text-overflow` or what-have-you.

**Comment by indigoxela** [comment]
Maybe this issue makes more sense as a "Task" and I'm pretty sure, it needs more feedback.

FTR: the realname module does no truncation and seems to have no problem with it.
